src/components/ModalPortal.js

```diff
@@ -130,11 +131,14 @@ export default class ModalPortal extends Component {
 
   beforeClose() {
     const { appElement, ariaHideApp, bodyOpenClassName } = this.props;
-    refCount.remove(bodyOpenClassName);
-    // Remove class if no more modals are open
-    if (refCount.count(bodyOpenClassName) === 0) {
-      elementClass(document.body).remove(bodyOpenClassName);
-    }
+    const classListMap = refCount.get();
+    // Decrement class(es) from the refCount tracker
+    // and remove unused class(es) from body
+    bodyOpenClassName
+      .split(' ')
+      .map(refCount.remove)
+      .filter(bodyClass => classListMap[bodyClass] === 0)
+      .forEach(bodyClass => document.body.classList.remove(bodyClass));
```

Comment:
Sweet! Any preference on the name? I was thinking something like `bodyManager` or `bodyElementManager`, but am definitely open to other suggestions.